src/test/groovy/com/pszymczyk/consul/infrastructure/ConsulWaiterTest.groovy

```diff
@@ -14,4 +18,80 @@ class ConsulWaiterTest extends Specification {
         def ex = thrown EmbeddedConsulException
         ex.message =~ "Could not start Consul*"
     }
+
+
+    def "should use token to get list of registered nodes"() {
```

Comment:
Another thing, why you add tests to `ConsulWaiterTest`? IMHO these test looks like tests for whole integration not for `CounsulWaiter`